From 77d2c94b4d5f2ef86fbd7862c55327cf88a200ad Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Sun, 30 Apr 2017 21:11:04 +0300
Subject: [PATCH 1/1] Introduce sane install recipe for Makefile

---
 Makefile | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index d9a7220..81677de 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,8 @@
 .SECONDEXPANSION:
 
+PREFIX=/usr
+LIBDIR=$(PREFIX)/lib64
+
 NPM_VERSION=4.2.0
 NODE_VERSION=7.8.0
 
@@ -199,9 +202,12 @@ build: $(WORKSPACE)/bin/heroku $(WORKSPACE)/lib/npm $(WORKSPACE)/lib/node $(WORK
 
 .PHONY: install
 install: build
-	cp -r $(WORKSPACE) /usr/local/lib/heroku
-	rm /usr/local/bin/heroku
-	ln -s /usr/local/lib/heroku/bin/heroku /usr/local/bin/heroku
+	install -dm00644 $(DESTDIR)$(PREFIX)/bin
+	install -dm00644 $(DESTDIR)$(LIBDIR)
+	cp -r $(WORKSPACE) $(DESTDIR)$(LIBDIR)/heroku
+	ln -s $(LIBDIR)/heroku/bin/heroku $(DESTDIR)$(PREFIX)/bin/heroku
+	rmdir $(DESTDIR)$(LIBDIR)/heroku/etc
+	rm -r $(DESTDIR)$(LIBDIR)/heroku/lib/{node,npm}
 
 .PHONY: clean
 clean:
-- 
2.12.2

